### DISCUSSION TITLE
Window borders and popup menus

### DISCUSSION THREAD

**Discussion by Avoncadoo** [prasanthrangan/hyprdots#1716] [Q&A] [answered]
Hey there, I had 2 questions:

1. How could one go about removing the borders between application windows? 

2. When I open up menus in certain apps like libre writer, instead of the standard pop it opens with weird behaviour (going off screen) How can I make apps use their default menus instead of the hyDE ones?

**Comment by OperatorRyu** [comment]
Hi there,

Hope you're having an awesome day, just to gather some more info here to help everyone understand what's going on:

with regards to "1" are you referring to general borders? (like the ones that contain "file", "view", etc etc. in electron applications) or the hyprland ones?

for 2, you might want to have a look at window behavioural rules in hyprland and tailor the window behaviour per application. Maybe give this a shot? https://wiki.hyprland.org/Configuring/Window-Rules/

Lemme know

**Comment by Avoncadoo** [comment]
for question 1, I'm referring to this:
![image](https://github.com/user-attachments/assets/5d574211-3e9a-4c9e-88e5-cef2ddcf2693)
(The blank space between my windows and between the windows and the borders of my screen). I'd rather have it so when the windows tile there is no space between them.

For question 2, I can't quite get a screenshot of the actual menus since they disappear whenever I try to take a screenshot, but basically whenever I hit one of these:
![image](https://github.com/user-attachments/assets/9d7b7c6c-1aaa-49a8-b5b9-2eb5cb7bab22)
it opens up a window that is floating and has rounded borders. You can also see that the app itself seems to be using the theme, which is what I want to disable.

Cheers!

**Reply by OperatorRyu** [reply, marked as answer]
For 1,

I don't see any general rules for this setup in the dotfiles so lets assume you're using Vim and you haven't made any changes to the config files yourself,

Back up  "/home/[your awesome username goes here]/.config/hypr
/hyprland.conf" in case you make a screw up and want to revert back.

Then run this in your terminal:
`sudo vim /home/[your awesome username goes here]/.config/hypr
/hyprland.conf`

Scroll down to "Layouts" (author didn't make a section for it so "layouts" is probably good) 
Add the following:
`general {
	gaps_in = 1 #change until comfy
	gaps_out = 1 #change until comfy
	layout = dwindle
	allow_tearing = false
}`

That should give you a more tight fit. You can add other things in the general object too if you like for example "border_size = 1" sets the size of the border to 1px and "col.active_border = (hex colour code)" sets the colour of the active border.

for 2, 
that is actually some thing I want to enable on my side haha, hopefully someone here can assist us both with that.

**Reply by Avoncadoo** [reply]
Thank you so much! After reading this I just looked around a bit and found where gaps are set- it's inside .config/hypr/themes/themes.conf. After changing it there my first issue was solved. Only thing is that when I change the theme it changes that as well, but I'm sure that issue can also be solved by going through the scripts.

**Reply by kRHYME7** [reply]
Hyprland configs can be overridden. So add your gaps configuration  inside `./userprefs.conf`

**Comment by kRHYME7** [comment]
2. @OperatorRyu  @Avoncadoo 

* What is in the screenshot is using the gtk theme
![image](https://github.com/user-attachments/assets/f964d0c6-3a24-4ad4-a83f-43816dba9f4e)
* Some apps like in the screenshot follows the gtk-theme being set. Some can be manually change in the apps setting depending on the app  but mostly they all just follow the gtk theme. 
* We cannot disable gtk theming (It fallsback to default white)   

* Solution will be selecting a good theme for this or check if the app is capable of setting its own theme.

**Reply by Avoncadoo** [reply]
Thank you very much for your answer, I'll try playing around with it.

**Reply by OperatorRyu** [reply]
Its the day that just keeps on giving! Thank you!
